xbmc/utils/LangCodeExpander.cpp

```diff
@@ -480,25 +482,26 @@ bool CLangCodeExpander::CompareFullLanguageNames(const std::string& lang1, const
 std::vector<std::string> CLangCodeExpander::GetLanguageNames(LANGFORMATS format /* = CLangCodeExpander::ISO_639_1 */, bool customNames /* = false */)
 {
   std::vector<std::string> languages;
-  const LCENTRY *lang = g_iso639_1;
-  size_t length = sizeof(g_iso639_1);
+
   if (format == CLangCodeExpander::ISO_639_2)
   {
-    lang = g_iso639_2;
-    length = sizeof(g_iso639_2);
+    for (const auto& codes : g_iso639_2)
+    {
+      languages.emplace_back(codes.name);
```

Comment:
Minor nitpick: You could use std::copy here and further down, would make the function even shorter